lib/deimos/config/phobos_config.rb

```diff
@@ -0,0 +1,132 @@
+        if f == :backoff && val
+          [:backoff, {
+            min_ms: val[0],
+            max_ms: val[-1]
+          }]
+        elsif val.present?
+          [f, val]
+        end
+      }.to_h
+    end
+
+    # :nodoc:
+    def reset!
+      super
+      Phobos.configure(self.phobos_config)
+    end
+
+    # Create a hash representing the config that Phobos expects.
+    # @return [Hash]
+    def phobos_config
+      p_config = {
+        logger: Logger.new(STDOUT),
+        custom_logger: self.phobos_logger,
+        custom_kafka_logger: self.kafka.logger,
+        kafka: {
+          client_id: self.kafka.client_id,
+          connect_timeout: self.kafka.connect_timeout,
+          socket_timeout: self.kafka.socket_timeout,
+          ssl_verify_hostname: self.kafka.ssl.verify_hostname,
+          seed_brokers: Array.wrap(self.kafka.seed_brokers)
+        },
+        producer: {
+          ack_timeout: self.producers.ack_timeout,
+          required_acks: self.producers.required_acks,
+          max_retries: self.producers.max_retries,
+          retry_backoff: self.producers.retry_backoff,
+          max_buffer_size: self.producers.max_buffer_size,
+          max_buffer_bytesize: self.producers.max_buffer_bytesize,
+          compression_codec: self.producers.compression_codec,
+          compression_threshold: self.producers.compression_threshold,
+          max_queue_size: self.producers.max_queue_size,
+          delivery_threshold: self.producers.delivery_threshold,
+          delivery_interval: self.producers.delivery_interval
+        },
+        consumer: {
+          session_timeout: self.consumers.session_timeout,
+          offset_commit_interval: self.consumers.offset_commit_interval,
+          offset_commit_threshold: self.consumers.offset_commit_threshold,
+          heartbeat_interval: self.consumers.heartbeat_interval
+        },
+        backoff: {
+          min_ms: self.consumers.backoff.to_a[0],
+          max_ms: self.consumers.backoff.to_a[-1]
+        }
+      }
+
+      p_config[:listeners] = self.consumer_objects.map do |consumer|
+        hash = consumer.to_h.reject do |k, _|
+          %i(class_name schema namespace key_config backoff).include?(k)
+        end
+        hash = hash.map { |k, v| [k, v.is_a?(Symbol) ? v.to_s : v] }.to_h
+        hash[:handler] = consumer.class_name
+        if consumer.backoff
+          hash[:backoff] = {
+            min_ms: consumer.backoff.to_a[0],
+            max_ms: consumer.backoff.to_a[-1]
+          }
+        end
+        hash
+      end
+
+      if self.kafka.ssl.enabled
+        %w(ca_cert client_cert client_cert_key).each do |key|
+          next if self.kafka.ssl.send(key).blank?
+
+          p_config[:kafka]["ssl_#{key}".to_sym] = ssl_var_contents(self.kafka.ssl.send(key))
+        end
+      end
+      p_config
+    end
+
+    # @param key [String]
+    # @return [String]
+    def ssl_var_contents(key)
+      File.exist?(key) ? File.read(key) : key
```

Comment:
Feels a little iffy mixing certs and file paths like this. I feel like they should be more explicitly separated, e.g. with a `file://` scheme or something. This feature wasn't supported in the old configs right?

Reply:
It was, actually. :) Truth is I can deprecate these since RubyKafka itself has separate configs for files or strings. Always felt a bit weird.

Reply:
Hm went back and checked and it only supports file path for ca_cert, not for the other options. Let's leave it as is for now.